hikari/interactions/command_interactions.py

```diff
@@ -370,3 +346,114 @@ def get_guild(self) -> typing.Optional[guilds.GatewayGuild]:
+            `CommandInteraction.create_initial_response` should be used to set
+            the interaction response message.
+
+        Examples
+        --------
+        ```py
+        async def handle_command_interaction(interaction: CommandInteraction) -> InteractionMessageBuilder:
+            return (
+                interaction
+                .build_response()
+                .add_embed(Embed(description="Hi there"))
+                .set_content("Konnichiwa")
+            )
+        ```
+
+        Returns
+        -------
+        hikari.api.special_endpoints.InteractionMessageBuilder
+            Interaction message response builder object.
+        """
+        return self.app.rest.interaction_message_builder(base_interactions.ResponseType.MESSAGE_CREATE)
+
+    def build_deferred_response(self) -> special_endpoints.InteractionDeferredBuilder:
+        """Get a deferred message response builder for use in the REST server flow.
+
+        !!! note
+            For interactions received over the gateway
+            `CommandInteraction.create_initial_response` should be used to set
+            the interaction response message.
+
+        !!! note
+            Unlike `hikari.api.special_endpoints.InteractionMessageBuilder`,
+            the result of this call can be returned as is without any modifications
+            being made to it.
+
+        Returns
+        -------
+        hikari.api.special_endpoints.InteractionMessageBuilder
+            Deferred interaction message response builder object.
+        """
+        return self.app.rest.interaction_deferred_builder(base_interactions.ResponseType.DEFERRED_MESSAGE_CREATE)
+
+
+@attr_extensions.with_copy
+@attr.define(hash=True, kw_only=True, weakref_slot=False)
+class AutocompleteInteraction(BaseCommandInteraction):
+    """Represents an autocomplete interaction on Discord."""
+
+    options: typing.Optional[typing.Sequence[AutocompleteInteractionOption]] = attr.field(
+        eq=False, hash=False, repr=True
+    )
+    """Parameter values provided by the user invoking this command."""
+
+    def build_response(
+        self, choices: typing.Sequence[commands.CommandChoice]
+    ) -> special_endpoints.InteractionAutocompleteBuilder:
+        """Get a message response builder for use in the REST server flow.
+
+        !!! note
+            For interactions received over the gateway
+            `AutocompleteInteraction.create_response` should be used to set
+            the interaction response.
+
+        Examples
+        --------
+        ```py
+        async def handle_autocomplete_interaction(interaction: AutocompleteInteraction) -> InteractionAutocompleteBuilder:
+            return (
+                interaction
+                .build_response(
+                    [
+                        OptionChoice(name="foo", value="a"),
+                        OptionChoice(name="bar", value="b"),
+                        OptionChoice(name="baz", value="c"),
```

Comment:
🙃 
```suggestion
                        CommandChoice(name="foo", value="a"),
                        CommandChoice(name="bar", value="b"),
                       CommandChoice(name="baz", value="c"),
```

Reply:
Yup, that's my bad. Got it from disnake.